addons/auth_oauth/views/auth_oauth_templates.xml

```diff
@@ -2,30 +2,36 @@
 <odoo>
         <template id="providers" name="OAuth Providers">
             <t t-if="len(providers) &gt; 0">
-                <em t-attf-class="d-block text-center text-muted small my-#{len(providers) if len(providers) &lt; 3 else 3}">- or -</em>
-                <div class="o_auth_oauth_providers list-group mt-1 mb-1 text-start">
-                    <a t-foreach="providers" t-as="p" class="list-group-item list-group-item-action py-2" t-att-href="p['auth_link']">
-                        <i t-att-class="p['css_class']"/>
-                        <t t-esc="p['body']"/>
-                    </a>
-                </div>
+                <a t-foreach="providers" t-as="p" class="list-group-item list-group-item-action py-2" t-att-href="p['auth_link']">
+                    <i t-att-class="p['css_class']"/>
+                    <t t-esc="p['body']"/>
+                </a>
             </t>
         </template>
 
         <template id="login" inherit_id="web.login" name="OAuth Login buttons">
-            <xpath expr="//div[hasclass('o_login_auth')]" position="inside">
+            <xpath expr="//div[hasclass('o_login_auth')]" position="attributes">
+                <attribute name="t-if">true</attribute>
+            </xpath>
+            <xpath expr="//div[hasclass('o_login_auth')]/div" position="inside">
```

Comment:
I am probably missing something but instead of this whole "`t-if="false"` being replaced by a `t-if="true"`" system, why can't you just add the whole

```xml
<div class="o_login_auth">
    <em class="d-block my-3 small text-center text-muted">- or -</em>
    <div class="o_auth_oauth_providers list-group"/>
</div>
```
section here? :thinking: And in the end, come back to what there was before and make it part of the "providers" template so that this code is not duplicated multiple times? :thinking: 

Not printing the empty `o_login_auth` div when there is no provider is good but I don't understand why it cannot just be added in the providers template to do that? :thinking:

Reply:
It's because there could be a passkey, which should render the or separator as well. I didn't find a better way to render the separator only once if there is either a passkey or a provider.

Reply:
Mmh ok :grimacing: I'll see if there is another way but maybe not indeed :+1: